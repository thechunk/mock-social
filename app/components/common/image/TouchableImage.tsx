import * as React from 'react';
import {PureComponent} from "react";
import {ImageProps, TouchableOpacity, TouchableOpacityProps} from "react-native";
import styles from './styles';
import {BlankImageUri} from "../../../styles/global";
import LoadedImage from "./LoadedImage";

export default class TouchableImage extends PureComponent<ImageProps & TouchableOpacityProps> {
    render() {
        return (
            <TouchableOpacity {...this.props as TouchableOpacityProps} style={this.props.style}>
                <LoadedImage
                    {...this.props as ImageProps}
                    style={styles.image}
                    showSpinner={true}
                    defaultSource={{uri: BlankImageUri}} />
            </TouchableOpacity>
        )
    }
}