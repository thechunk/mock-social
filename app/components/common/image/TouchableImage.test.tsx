import * as React from 'react';
import {shallow} from "enzyme";
import TouchableImage from "./TouchableImage";

describe('TouchableImage', () => {
    it('renders correctly', () => {
        expect(shallow(<TouchableImage source={{uri: 'uri'}}/>)).toMatchSnapshot();
    });
});