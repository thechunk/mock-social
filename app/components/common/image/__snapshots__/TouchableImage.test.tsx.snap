// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TouchableImage renders correctly 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  source={
    Object {
      "uri": "uri",
    }
  }
>
  <LoadedImage
    defaultSource={
      Object {
        "uri": "data:image/gif;base64,R0lGODlhAQABAIAAAP///wAAACwAAAAAAQABAAACAkQBADs=",
      }
    }
    showSpinner={true}
    source={
      Object {
        "uri": "uri",
      }
    }
    style={
      Object {
        "height": "100%",
        "width": "100%",
      }
    }
  />
</TouchableOpacity>
`;
